plugins/modules/k8s.py

```diff
@@ -172,6 +172,20 @@
         - When set to True, server-side apply will force the changes against conflicts.
         type: bool
         default: False
+  select_all:
+    description:
+    - When this option is set to I(true), action will be performed on all resources, in the namespace of the specified resource type.
+    - When I(select_all=True) and I(state=absent), C(kind) is required.
+    - When I(select_all=True) and I(state=patched) a valid YAML definition must be provided using option C(src) or C(resource_definition),
+      resource type must be defined using C(kind) option or into provided YAML definition.
+    - Ignored when I(state=present).
+    - Ignored when I(state=absent) and one of C(src), C(name) or C(resource_definition) is provided.
+    - Ignored when I(state=patched) and C(name) is provided or C(src) or C(resource_definition) is provided with a list containing
+      more than one YAML definition, or if the resource definition contains a metadata.name field.
```

Comment:
I am not sure about this user experience. This feels like a significant amount of complexity stuffed into a single option, which is the kind of thing that makes me think we may be approaching this the wrong way. I don't have a better suggestion at the moment, I'd need to think some more about this. Would be interested to see if @tima has any thoughts.

Reply:
I understand your point, we can limit this option to `state=absent` to easy user experience